### PR TITLE
SNAT improvemets

### PR DESCRIPTION
**Pull request by andre8244**
- While deleting a firewall object, ensure it is not used as SNAT source object
- Add inline documentation

NethServer/dev#6457

### CODE REVIEW

File: ui/public/i18n/language.json [hunk 1]
```diff
@@ -588,6 +588,8 @@
...
         "rules_service": "Enter the name of a network service running on the firewall, otherwise leave empty to match any service",
         "allow_only": "Restrict access to port forward only from a list of public IP addresses or networks (written in CIDR format). Add one entry per line. Leave the field empty to allow access by anyone.",
-        "SipAlg": "ALG allows SIP and H.323 protocols to operate through NAT. When enabled, ALG will inspect and rewrite Voice over Internet Protocol (VoIP) network packets and open required ports. Try to disable this option only if you're experiencing audio and call problems with your PBX or VoIP client."
+        "SipAlg": "ALG allows SIP and H.323 protocols to operate through NAT. When enabled, ALG will inspect and rewrite Voice over Internet Protocol (VoIP) network packets and open required ports. Try to disable this option only if you're experiencing audio and call problems with your PBX or VoIP client.",
+        "more_info_about": "More info about",
+        "snat_order": "The following entries are order-sensitive. The first entry that matches a particular connection will be the one that is used"
```
Comment (edospadoni) [inline ui/public/i18n/language.json:593]:
```suggestion
        "snat_order": "The following entries are order-sensitive. The first entry that matches a particular connection will be the one that is used."
```

**Review by edospadoni — APPROVED** [review]

File: NethServer/Firewall.pm [hunk 2]
```diff
@@ -857,6 +857,19 @@ sub getPortForwards
...
+Each record has all database properties.
+
+=cut
+sub getAliases
+{
+    my $self = shift;
+    my @list = $self->{'ndb'}->aliases;
+    return @list;
+}
+
```
Comment (gsanchietti) [inline NethServer/Firewall.pm:872]: This function is built-in inside NetworksDB:
```
perldoc esmith::NetworksDB
```

File: NethServer/Firewall.pm [hunk 3]
```diff
@@ -962,6 +975,7 @@ The object is searched inside one of following lists:
 * proxy bypasses
 * traffic shaping rules
 * port forwards
+* interface aliases
```
Comment (gsanchietti) [inline NethServer/Firewall.pm:978]:
```suggestion
* source nat rules
```

File: NethServer/Firewall.pm [hunk 4]
```diff
@@ -1013,12 +1027,14 @@ sub countReferences($$)
     my @tcrules = $self->getTcRules(); 
     my @pfrules = $self->getPortForwards();
     my @bypass = $self->getBypassRules();
+    my @aliases = $self->getAliases();
```
Comment (gsanchietti) [inline NethServer/Firewall.pm:1030]:
```suggestion
    my @aliases = $self->{'ndb'}->aliases;
```

File: NethServer/Firewall.pm [hunk 5]
```diff
@@ -1013,12 +1027,14 @@ sub countReferences($$)
...
     push(@fwrules, @tcrules);
     push(@fwrules, @pfrules);
     push(@fwrules, @bypass);
+    push(@fwrules, @aliases);
```
Comment (gsanchietti) [inline NethServer/Firewall.pm:1034]: This is a bit misleading.

Please separate the logic for aliases by creating a new loop after the one `@fwrules` array.

**Review by gsanchietti — CHANGES_REQUESTED** [review]

**Review by gsanchietti — APPROVED** [review]

**Pull request merged**